docs/modules/array.md

```diff
@@ -11,11 +11,88 @@ Arrays can be declared as constants in a flowgraph or variables in
 
 The Array module exposes a limited set of operations on array variables.
 
-## Array.length 
+## Array.concat
+
+Combine the elements of a list of arrays into a single array
+```
+Array.concat(array1, array2, ...)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array1`    |   The first elements in the resulting array  | Yes
+`array2`    |   Elements to append to array1  | No
+`...`     | Any number of additional arrays | No
+
+_Example: Concatenate `["hello"]`, `[]` and `["world"]` into `["hello", "world"]`_
+
+```
+{!docs/examples/modules/array_concat.juttle!}
+```
+
+## Array.indexOf
+
+Search an array and return the zero-based index of the first match, or
+-1 if there is no match.
+
+```
+Array.indexOf(array, searchvalue)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    |   The array to search  | Yes
+`searchvalue`  |  The value for which to search  |  Yes
+
+_Example: Search for a random integer in an array of random integers_
+
+```
+{!docs/examples/modules/array_indexof.juttle!}
+```
+
+## Array.join
+
+Combine the elements of an array into a string
+```
+Array.join(array, separator)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    |   Array whose elements to join | Yes
+`separator`| Separator to delimit the joined elements  | No
+
+_Example: Join ['joe','meg','bob','may'] into the string 'joe,meg,bob,may'_
+
+```
+{!docs/examples/modules/array_join.juttle!}
+```
+
+## Array.lastIndexOf
+
+Search an array and return the zero-based index of the last match, or
+-1 if there is no match.
+
+```
+Array.lastIndexOf(array, searchvalue)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    |   The array to search  | Yes
+`searchvalue`  |  The value for which to search  |  Yes
+
+_Example: Search for a random integer in an array of random integers_
+
+```
+{!docs/examples/modules/array_indexof.juttle!}
```

Comment:
That example is not added to the PR as noted above, but here I think you wanted array_last_index_of.juttle which is added.